homeassistant/components/nibe_heatpump/climate.py

```diff
@@ -0,0 +1,349 @@
+    entity_description: ClimateDescription
+
+    def __init__(
+        self,
+        coordinator: Coordinator,
+        entity_description: ClimateDescription,
+    ) -> None:
+        """Initialize entity."""
+        super().__init__(
+            coordinator,
+            {
+                entity_description.current_address,
+                entity_description.setpoint_heat_address,
+                entity_description.setpoint_cool_address,
+                entity_description.prio_address,
+                entity_description.mixing_valve_state_address,
+                entity_description.active_accessory_address,
+                entity_description.use_room_sensor_address,
+                entity_description.cooling_with_room_sensor_address,
+            },
+        )
+        self._attr_available = False
+        self._attr_name = entity_description.name
+        self._attr_unique_id = f"{coordinator.unique_id}-{entity_description.key}"
+        self._attr_device_info = coordinator.device_info
+        self._attr_hvac_action = HVACAction.IDLE
+        self._attr_hvac_mode = HVACMode.OFF
+        self._attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF, HVACMode.HEAT]
+        self._attr_supported_features = (
+            ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+            | ClimateEntityFeature.TARGET_TEMPERATURE
+        )
+        self._attr_target_temperature_high = None
+        self._attr_target_temperature_low = None
+        self._attr_target_temperature = None
+        self._attr_target_temperature_step = 0.5
+        self._attr_entity_registry_enabled_default = (
+            entity_description.active_accessory_address is None
+        )
+
+        self._attr_max_temp = 35.0
+        self._attr_min_temp = 5.0
+
+        def _get(address: int) -> Coil:
+            return coordinator.heatpump.get_coil_by_address(address)
+
+        self._coil_current = _get(entity_description.current_address)
+        self._coil_setpoint_heat = _get(entity_description.setpoint_heat_address)
+        self._coil_setpoint_cool = _get(entity_description.setpoint_cool_address)
+        self._coil_prio = _get(entity_description.prio_address)
+        self._coil_mixing_valve_state = _get(
+            entity_description.mixing_valve_state_address
+        )
+        if entity_description.active_accessory_address is None:
+            self._coil_active_accessory_address = None
+        else:
+            self._coil_active_accessory_address = _get(
+                entity_description.active_accessory_address
+            )
+        self._coil_use_room_sensor = _get(entity_description.use_room_sensor_address)
+        self._coil_cooling_with_room_sensor = _get(
+            entity_description.cooling_with_room_sensor_address
+        )
+
+        if self._coil_current:
+            self._attr_temperature_unit = self._coil_current.unit
+
+    def _handle_coordinator_update(self) -> None:
+        if not self.coordinator.data:
```

Comment:
This might be a bit more readable if you did 

`coordinator = self.coordinator` at the top of the function.

It would also save quite a few lookups (not that performance is going to be an issue here)